TestingLowerBounds/FDiv/Basic.lean

```diff
@@ -728,6 +765,29 @@ lemma fDiv_nonneg [IsProbabilityMeasure μ] [IsProbabilityMeasure ν]
   calc (0 : EReal) = f (μ Set.univ).toReal := by simp [hf_one]
   _ ≤ fDiv f μ ν := le_fDiv hf_cvx hf_cont
 
+/- The hypothesis `hfg'` can maybe become something like `f ≤ᵐ[atTop] g`, but then we would need
+some lemma like `derivAtTop_mono`, and I'm not sure this is true in gneral, without any assumption on `f`.
+We could prove it if we had some lemma saying that the new derivAtTop is equal to the old definition,
+this is probably false in general, but under some assumptions it should be true. -/
+lemma fDiv_mono' (hf_int : Integrable (fun x ↦ f ((∂μ/∂ν) x).toReal) ν)
+    (hfg : f ≤ᵐ[ν.map (fun x ↦ ((∂μ/∂ν) x).toReal)] g) (hfg' : derivAtTop f ≤ derivAtTop g) :
+    fDiv f μ ν ≤ fDiv g μ ν := by
+  rw [fDiv_of_integrable hf_int, fDiv]
+  split_ifs with hg_int
+  swap; · simp
+  gcongr
+  · exact EReal.coe_le_coe_iff.mpr <| integral_mono_ae hf_int hg_int <|
+      ae_of_ae_map (Measure.measurable_rnDeriv μ ν).ennreal_toReal.aemeasurable hfg
+  · exact EReal.coe_ennreal_nonneg _
+
+lemma fDiv_mono (hf_int : Integrable (fun x ↦ f ((∂μ/∂ν) x).toReal) ν)
+    (hfg : f ≤ g) (hfg' : derivAtTop f ≤ derivAtTop g) : fDiv f μ ν ≤ fDiv g μ ν :=
+  fDiv_mono' hf_int (eventually_of_forall hfg) hfg'
```

Comment:
`derivAtTop_mono` is true for convex functions, which are the only functions for which fDiv really makes sense. The name `fDiv_mono` should hence be reserved for the lemma that assumes convex functions and does not need `hfg'`. Can you change the name of this one and add a TODO about that?

Also I think `hf_int` can be removed if both function are a.e. strongly measurable and convex, since then `g` would not be integrable either and the `fDiv` on the right would be infinite. Don't change anything now, but please write a comment about it.

Reply:
Okay, to remove the integrability condition I think we need to ask also for the finiteness of the measures, otherwise we have a counterexample with `f = -1` and `g = 0`.